drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -476,12 +791,12 @@ s32_t stm32_i2c_msg_read(struct device *dev, struct i2c_msg *msg,
 			if (LL_I2C_IsActiveFlag_AF(i2c) || stm32_i2c_wait_timeout(&timeout)) {
 				LL_I2C_ClearFlag_AF(i2c);
 				LL_I2C_GenerateStopCondition(i2c);
-				LOG_DBG("%s: NACK", __func__);
-
-				return -EIO;
+				data->current.is_nack = 1U;
+				goto end;
 			}
 		}
 
+		LL_I2C_ClearFlag_ADDR(i2c);
```

Comment:
Is that change related to your addition?

Reply:
If not please do this as a separate commit.

Reply:
Without this master read operation for 2 bytes will generate nack after first byte, thus last one wont be sent by the slave. So it's definitely required for proper work of i2c master. Just tell me, what should i do

Reply:
Ok, then maybe add a comment in code to state this clearly.

Reply:
done